### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by munderwoods**
Took out simple-cdd and changed username to pi

### CODE REVIEW

**Comment by munderwoods** [comment]
- docker is redundant now
- some missing setup info if you want to stay with Docker
- debian subfolder
- sudoers changes
- remove redundant sudo group change
- debian mirror redirector dl
- output artifact perms

File: README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+4. Run "make" to see a list of available procedures.
+5. Run "make init" to build the docker container and perform initial setup.
+6. Open the DEBIAN_VERSION file and change it to whatever version you'd like.
+7. Edit the preseed file however you'd like (https://wiki.debian.org/DebianInstaller/Preseed)
+8. Run the make command for the architecture you desire (example: "make build-amd64") or "make build-all" to build for all architectures.
+9. After building, your ISO(s) will be written to the dists directory and prepended with the name "retronas" and the date.
+
+## How it works:
+
+The Makefile just runs the bash scripts in the repo with different arguments depending on what you want to do. Different distributions are split up into their own directories.
```
Comment (sairuk) [inline README.md:42]: remove the word `just`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Every build command has as its dependency a download command. This just looks for the ISO for the specific architecture and version in the appropriate iso-cache directory. If it finds it, then this step is skipped. If not, it downloads the appropriate ISO to that dist's iso-cache directory. 
+
+### Debian
+
+After the ISO is downloaded, the build command is run for the specified architecture. The iso iss unpacked into a temporary directory called isofiles. The file "preseed.cfg" is then injected into that ISO's initrd file. Initrd  handles the installation of Debian. During the installation it always looks for a file named preseed.cfg for configuration, but it usually doesn't exist so it just performs its default behavior.
```
Comment (sairuk) [inline README.md:48]: typo `iss`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+The directory "isofiles" is then packed back up into a new RetroNAS image that you can boot.
```
Comment (sairuk) [inline README.md:50]: remove `back up` ... `packed into` is fine

**Review by sairuk — CHANGES_REQUESTED** [review]
Looks good overall, please see inline notes

File: README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+### Building your own image:
+RetroNAS intends to maintain a release for whatever the current stable version of Debian is. At the time of this writing, that's Bullseye 11.2.0. If, however, you wish to build an installer with a different version, or to edit the preseed, the procedure is quite simple:
+
+1. Install the dependencies: make, bsdtar(libarchive-tools), cpio, xorriso and curl
+2. Clone this repo.
+3. Navigate to the retronas-dist directory.
+4. Run "make" to see a list of available procedures.
+5. Run "make init" to build the docker container and perform initial setup.
+6. Open the DEBIAN_VERSION file and change it to whatever version you'd like.
```
[the remark this answers is not in the document]
Comment (sairuk) [reply README.md:35]: make init is no longer valid

File: README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+7. Edit the preseed file however you'd like (https://wiki.debian.org/DebianInstaller/Preseed)
+8. Run the make command for the architecture you desire (example: "make build-amd64") or "make build-all" to build for all architectures.
```
Comment (sairuk) [inline README.md:37]: this is now called `build-debian-amd64`

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,31 @@
+.DEFAULT_GOAL := help
+DEBIAN_VERSION=$(shell cat debian/DEBIAN_VERSION)
+
+LOCAL_DIR = $(shell pwd)
+
+
+make-dists: ## Make the directory where the preseeded distrobution images will end up
```
[the remark this answers is not in the document]
Comment (munderwoods) [reply Makefile:7]: Then why they always call them distros?

File: debian/preseed.cfg [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+d-i mirror/http/proxy string
+
+
+# Users/passwords
+d-i passwd/root-login boolean true
+d-i passwd/make-user boolean true
+d-i passwd/root-password password retronas
+d-i passwd/root-password-again password retronas
+d-i passwd/user-fullname string retronas
+d-i passwd/username string pi
```
[the remark this answers is not in the document]
Comment (munderwoods) [reply debian/preseed.cfg:42]: Yeah lemme know what you wanna do about that whenever you're ready.

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (sairuk) [reply Makefile]: that is a question for the larger linux community, correct spelling is `distribution`

**Pull request merged**